src/bucket/Bucket.cpp

```diff
@@ -854,4 +854,28 @@ Bucket::getBucketVersion(std::shared_ptr<Bucket const> const& bucket)
     BucketInputIterator it(bucket);
     return it.getMetadata().ledgerVersion;
 }
+
+BucketEntryCounters
+Bucket::getBucketEntryCounters() const
+{
+    if (mIndex)
```

Comment:
I think this should probably just `releaseAssert(mIndex)`. The reason being if we return empty here, downstream might run into confusing bugs where metrics are returning as all zeros because of a misconfiguration, and it's not clear to downstream that these metrics would be tied to SQL database config flags. I would instead assert and protect calls to the metrics endpoint with `isUsingBucketListDB` checks.